src/main/java/org/jabref/logic/importer/fileformat/BibliographyFromPdfImporter.java

```diff
@@ -0,0 +1,301 @@
+        } catch (EncryptedPdfsNotSupportedException e) {
+            return ParserResult.fromErrorMessage(Localization.lang("Decryption not supported."));
+        } catch (IOException exception) {
+            return ParserResult.fromError(exception);
+        }
+
+        ParserResult parserResult = new ParserResult(result);
+
+        // Generate citation keys for result
+        CitationKeyGenerator citationKeyGenerator = new CitationKeyGenerator(parserResult.getDatabaseContext(), citationKeyPatternPreferences);
+        parserResult.getDatabase().getEntries().forEach(citationKeyGenerator::generateAndSetKey);
+
+        return parserResult;
+    }
+
+    private record IntermediateData(String number, String reference) {
+    }
+
+    /**
+     * In: <code>"[1] ...\n...\n...[2]...\n...\n...\n[3]..."</code><br>
+     * Out: <code>List&lt;String> = ["[1] ...", "[2]...", "[3]..."]</code>
+     */
+    private List<BibEntry> getEntriesFromPDFContent(String contents) {
+        List<IntermediateData> referencesStrings = new ArrayList<>();
+        Matcher matcher = REFERENCE_PATTERN.matcher(contents);
+        while (matcher.find()) {
+            String reference = matcher.group(2).replaceAll("\\r?\\n", " ").trim();
+            referencesStrings.add(new IntermediateData(matcher.group(1), reference));
+        }
+
+        return referencesStrings.stream()
+                                .map(data -> parseReference(data.number(), data.reference()))
+                                .toList();
+    }
+
+    private String getLastPageContents(PDDocument document) throws IOException {
+        PDFTextStripper stripper = new PDFTextStripper();
+
+        int lastPage = document.getNumberOfPages();
+        stripper.setStartPage(lastPage);
+        stripper.setEndPage(lastPage);
+        StringWriter writer = new StringWriter();
+        stripper.writeText(document, writer);
+
+        return writer.toString();
+    }
+
+    /**
+     * Example: <code>J. Knaster et al., “Overview of the IFMIF/EVEDA project”, Nucl. Fusion, vol. 57, p. 102016, 2017. doi:10.1088/ 1741-4326/aa6a6a</code>
+     *
+     * @param number     The number of the reference - used for logging only
+     */
+    @VisibleForTesting
+    BibEntry parseReference(String number, String reference) {
+        String originalReference = "[" + number + "] " + reference;
+        BibEntry result = new BibEntry(StandardEntryType.Article);
+
+        reference = reference.replace(".-", "-");
+
+        // J. Knaster et al., “Overview of the IFMIF/EVEDA project”, Nucl. Fusion, vol. 57, p. 102016, 2017. doi:10.1088/ 1741-4326/aa6a6a
+        // Y. Shimosaki et al., “Lattice design for 5 MeV – 125 mA CW RFQ operation in LIPAc”, in Proc. IPAC’19, Mel- bourne, Australia, May 2019, pp. 977-979. doi:10.18429/ JACoW-IPAC2019-MOPTS051
+        int pos = reference.indexOf("doi:");
+        if (pos >= 0) {
+            String doi = reference.substring(pos + 4).trim();
```

Comment:
Sure that this are always 4 characters?

Reply:
I am pretty sure that the constant string "doi:" alwas has 4 characters. But in a parallel universe this might change. Thus, I will change to "doi:'.length() later